pr_toolbox/quantum/results/frequencies.py

```diff
@@ -38,27 +60,61 @@ def map_counts(counts: Counts, mapper: Callable) -> Counts:
     return Counts(counts_dict)
 
 
-def bitflip_counts(counts: Counts, bitflips: int) -> Counts:
-    """Flip readout bits in counts according to the input bitflips (int encoded).
+def map_quasi_dists(quasi_dists: QuasiDistribution, mapper: Callable) -> QuasiDistribution:
+    """Map quasi-distributions by reassigning keys according to input callable.
 
     Args:
-        counts: the counts to process.
+        quasi_dists: the quasi-distributions to process.
+        mapper: the callable to map readout bits (i.e. counts keys).
+
+    Returns:
+        New QuasiDistribution with readout bits mapped according to input callable.
+    """
+    counts_dict: dict[int, float] = defaultdict(lambda: 0)
+    for readout, freq in quasi_dists.items():
+        readout = mapper(readout)
+        counts_dict[readout] += freq
+    return QuasiDistribution(
+        counts_dict, shots=quasi_dists.shots, stddev_upper_bound=quasi_dists.stddev_upper_bound
+    )
+
+
+def bitflip_frequencies(frequencies: FrequenciesLike, bitflips: int) -> FrequenciesLike:
+    """Flip readout bits in frequencies according to the input bitflips (int encoded).
+
+    Args:
+        frequencies: the frequencies to process.
         bitflips: the bitflips to be applied.
 
     Returns:
-        New counts with readout bits flipped according to input.
+        New frequencies with readout bits flipped according to input.
     """
-    return map_counts(counts, lambda readout: readout ^ bitflips)
+    return map_frequencies(frequencies, lambda readout: readout ^ bitflips)
 
 
-def bitmask_counts(counts: Counts, bitmask: int) -> Counts:
-    """Apply mask to readout bits in counts.
+def bitmask_frequencies(frequencies: FrequenciesLike, bitmask: int) -> FrequenciesLike:
+    """Apply mask to readout bits in frequencies.
 
     Args:
-        counts: the counts to process.
+        frequencies: the frequencies to process.
         bitmask: the bit mask to be applied.
 
     Returns:
         New counts with readout bits masked according to input.
     """
-    return map_counts(counts, lambda readout: readout & bitmask)
+    return map_frequencies(frequencies, lambda readout: readout & bitmask)
+
+
+def convert_counts_to_quasi_dists(counts: Counts) -> QuasiDistribution:
+    """Infers a :class:`~qiskit.result.QuasiDistribution` from :class:`~qiskit.result.Counts`.
+
+    Args:
+        counts: the counts to convert.
+
+    Returns:
+        New QuasiDistribution inferred from counts.
+    """
+    shots = counts.shots()
+    std_dev = sqrt(1 / shots) if shots else None
+    probabilities = {k: v / (shots or 1) for k, v in counts.int_outcomes().items()}
+    return QuasiDistribution(probabilities, shots=shots, stddev_upper_bound=std_dev)
```

Comment:
Very good! But this should be a separate PR 😅  Two reasons for this:
1. It is an independent contribution doing its own thing, so splitting it up in two PRs will make this more apparent and ensure that it shows up as an entire new entry in the change log.
2. Will help keep this PR smaller.

Should be ready to merge if you port it (test looks good as well) 🙃

Reply:
Done! You can find the dedicated PR in #3 .